tests/petsc/petsc_ts_03.cc

```diff
@@ -136,7 +136,6 @@ class HarmonicOscillator
         // solver is used.
         time_stepper.explicit_function =
           [&](const real_type t, const VectorType &y, VectorType &res) -> void {
-          deallog << "Evaluating explicit function at t=" << t << std::endl;
```

Comment:
This and the other output log is on purpose. We want to know how often (and when) these two functions are called. Do you have a specific reason to remove these?

Reply:
The number of occurrences *is* different in the CI (and also locally) and I assumed the important out out is coming from
```C++
deallog << "Intermediate output:" << std::endl;
deallog << "  t =" << t << std::endl;
deallog << "  y =" << sol[0] << "  (exact: " << std::exp(-kappa * t)
<< ')' << std::endl;
```

Reply:
@masterleinad I'm also working on cleaning up these tests. If you want, I can add these changes to my branch if you tell me the specific compiler and petsc configuration it is triggering the problem

Reply:
I reverted the changes to these two tests for now.

Reply:
It's failing for me with PETSc 3.19.2.

Reply:
See https://cdash.dealii.org/testSummary.php?project=1&name=petsc%2Fpetsc_ts_03.debug&date=2023-06-10 for failing configurations.